typed-racket-lib/typed-racket/base-env/contract-prims.rkt

```diff
@@ -0,0 +1,346 @@
+#lang racket/base
+;; This module provides typed versions of identifiers from racket/contract (not
+;; to be confused with prims-contract).
+
+;; Many contract forms (e.g. >/c) can be directly ascribed a type, which is done
+;; using the define-contract macro. Others, though, have more complicated
+;; typings and so the typed version annotates the typechecking-relevant pieces
+;; of surface syntax before delegating to the untyped form. The annotated pieces
+;; are retrieved by the form's type rule, after the form has fully expanded.
+
+(require "../utils/utils.rkt"
+         (for-syntax racket/base
+                     racket/sequence
+                     racket/syntax
+                     racket/list
+                     syntax/parse
+                     syntax/transformer
+                     (utils contract-utils)
+                     (private syntax-properties))
+         ;; these types go into ignore-some-expr-property, which must contain
+         ;; syntax objects for surface-level types (e.g. Real instead of -Real)
+         (base-env base-types base-types-extra)
+         (prefix-in untyped: racket/contract/base))
+(provide (except-out (all-defined-out)
+                     define-contract)
```

Comment:
It might be nice to not use `(all-defined-out)`, and instead make `define-contract` export each form, and then just manually export the other forms.